javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -879,4 +893,65 @@ public static TypeMirror substitute(
                 com.sun.tools.javac.util.List.from(newP),
                 com.sun.tools.javac.util.List.from(newT));
     }
+
+    /**
+     * Returns a fresh type variable, a captued typed, for {@code typeMirror}, if {@code typeMirror}
+     * is a wildcard. If it is not a wildcard, {@code typeMirror} is returned
+     *
+     * @param typeMirror some type
+     * @param env processing environment
+     * @return a fresh type variable if {@code typeMirror} is a wildcard, or {@code type mirror}
+     */
+    public static TypeMirror freshTypeVariable(TypeMirror typeMirror, ProcessingEnvironment env) {
+        JavacProcessingEnvironment javacEnv = (JavacProcessingEnvironment) env;
+        com.sun.tools.javac.code.Types types =
+                com.sun.tools.javac.code.Types.instance(javacEnv.getContext());
+        return types.freshTypeVariables(com.sun.tools.javac.util.List.of((Type) typeMirror)).head;
+    }
+
+    /**
+     * Returns the list of type variables such that a type variable in the list only references type
+     * variables at a lower index than itself.
+     *
+     * @param collection a collection of type variables
+     * @param types types
+     * @return the list of type variables such that a type variable in the list only references type
+     *     variables at a lower index than itself
+     */
+    public static List<TypeVariable> order(Collection<TypeVariable> collection, Types types) {
+        List<TypeVariable> list = new ArrayList<>(collection);
+        List<TypeVariable> ordered = new ArrayList<>();
+        while (!list.isEmpty()) {
+            TypeVariable free = free(list, types);
+            list.remove(free);
+            ordered.add(free);
+        }
+        return ordered;
+    }
+
+    /**
+     * Returns the first TypeVariable in {@code collection} that does not contain any other type in
+     * the collection, but maybe its self.
+     *
+     * @param collection a collection of type variables
+     * @param types types
+     * @return the first TypeVariable in {@code collection} that does not contain any other type in
+     *     the collection, but maybe its self
+     */
+    @SuppressWarnings("interning:not.interned") // must be the same object from collection
+    public static TypeVariable free(Collection<? extends TypeVariable> collection, Types types) {
```

Comment:
Does this need to be public?  It feels like a helper method.
Also, I don't find the name `free` particularly evocative.